### PR TITLE
docs: add docs about list-like feature

### PR DESCRIPTION
**Pull request by dongxiang123**
Signed-off-by: dong xiang <iamxiangdong@gmail.com>

Goals:
Since this [PR](https://github.com/docarray/docarray/pull/730/files#diff-ed0be5f19866ce0e879e0ad77f486223c573998977944684dc03499250e5089b) is merge and we need to add document about constructing a new backend in the document
- ...
- ...
- [ ] check and update documentation, if required. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines)

### CODE REVIEW

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docs/advanced/document-store/extend.md [hunk 1]
```diff
@@ -84,7 +84,16 @@ You will need to implement the above five functions, which correspond to the log
...
 inherited from the `BaseGetSetDelMixin`. Therefore, you need to implement methods to persist this mapping, in case you 
-want to also persist the ordering of Documents inside the storage.
+want to also persist the ordering of Documents inside the storage. Considered the performance issue brought by list-like
+structure implemented by Offset2id, you can disable this feature by passing a flag when constructing the backend.
+After so, the Offset2id will become a shell and do nothing.
+
+```Python
+from docarray import DocumentArray, Document
+
+da = DocumentArray(storage='qdrant', config={'n_dim':3, 'list_like':False})
```
[the remark this answers is not in the document]
Comment (dongxiang123) [reply docs/advanced/document-store/extend.md:94]: Make sense

**Comment by JohannesMessner** [comment]
I think we need to explain that this `list_like` parameter has to be part of the `config`, and then how it is passed to the `__init__` of `Offsed2ID`.

**Comment by dongxiang123** [comment]
> I think we need to explain that this `list_like` parameter has to be part of the `config`, and then how it is passed to the `__init__` of `Offsed2ID`.

added examples.

File: docs/advanced/document-store/extend.md [hunk 2]
```diff
@@ -84,7 +84,19 @@ You will need to implement the above five functions, which correspond to the log
 
 Note that DocumentArray maintains an `offset2ids` mapping to allow a list-like behaviour. This mapping is 
 inherited from the `BaseGetSetDelMixin`. Therefore, you need to implement methods to persist this mapping, in case you 
-want to also persist the ordering of Documents inside the storage.
+want to also persist the ordering of Documents inside the storage. Considered the performance cost brought by list-like
+structure implemented by Offset2id, you can disable this feature by passing a flag when constructing the backend.
+After so, the Offset2id will become a shell and do nothing. Then in the `getsetdel.py` you can construct the _offset2ids 
+member variable by passing list_like flag as follows:
```
Comment (JohannesMessner) [inline docs/advanced/document-store/extend.md:90]:
```suggestion
want to also persist the ordering of Documents inside the storage. However, the list-like
structure implemented by Offset2id can introduce performance bottlenecks, so you can disable this feature by passing a flag when constructing the backend.

In your backend implementations, in `getsetdel.py`, you have to construct the _offset2ids 
member variable by passing list_like flag as follows:
```

File: docs/advanced/document-store/extend.md [hunk 3]
```diff
@@ -203,6 +225,17 @@ class BackendMixin(BaseBackendMixin):
 
 `MyDocStoreConfig` is a dataclass for containing the configs. You can expose arguments of your document store to this data class and allow users to customize them. In `init_storage` function, you need to parse `config` either from `MyDocStoreConfig` object or a `Dict`.
 
+The `list-like` flag is passed into the config file as follows:
```
Comment (JohannesMessner) [inline docs/advanced/document-store/extend.md:228]:
```suggestion
To enable the disabling of list-like features, your configuration should accept the flag `list_like` as follows:
```

File: docs/advanced/document-store/extend.md [hunk 4]
```diff
@@ -203,6 +225,17 @@ class BackendMixin(BaseBackendMixin):
...
+```python
+@dataclass
+class MyDocStoreConfig:
+    config1: str
+    config2: str
+    list - like: True
+    config3: Dict
+    ...
+```
+By default, it is set to be `True`.
```
Comment (JohannesMessner) [inline docs/advanced/document-store/extend.md:238]:
```suggestion
By default, this should be set to be `True`.

Further, you have to store the value of this flag in `self._list_like`. Some methods that are handled outside of your control will take the value form there and use it appropriately.
```

**Review by JohannesMessner — CHANGES_REQUESTED** [review]

File: docs/advanced/document-store/extend.md [hunk 5]
```diff
@@ -203,6 +225,17 @@ class BackendMixin(BaseBackendMixin):
 
 `MyDocStoreConfig` is a dataclass for containing the configs. You can expose arguments of your document store to this data class and allow users to customize them. In `init_storage` function, you need to parse `config` either from `MyDocStoreConfig` object or a `Dict`.
 
+The `list-like` flag is passed into the config file as follows:
+```python
+@dataclass
+class MyDocStoreConfig:
+    config1: str
+    config2: str
+    list - like: True
```
Comment (JoanFM) [inline docs/advanced/document-store/extend.md:234]:
```suggestion
    list_like: bool
```

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docs/advanced/document-store/extend.md (hunk 2 above)
Comment (alexcg1) [inline docs/advanced/document-store/extend.md:90]:
```suggestion
want to also persist the ordering of Documents inside the storage. Considering the performance cost brought by the list-like structure implemented by Offset2id, you can disable this feature by passing a flag when constructing the backend.
After this, the Offset2id will become a shell and do nothing. Then in the `getsetdel.py` you can construct the `_offset2ids` 
member variable by passing the `list_like` flag as follows:
```

File: docs/advanced/document-store/extend.md [hunk 6]
```diff
@@ -156,6 +168,16 @@ Most of the interfaces come from Python standard [MutableSequence](https://docs.
 As a reference, to see how we implement for SQLite, check out {class}`~docarray.array.storage.sqlite.seqlike.SequenceLikeMixin`.
 ```
 
+For the reason to support the list-like feature, the list-like APIs should do flag checking only when the offset2id structure is called as follows:
```
Comment (alexcg1) [inline docs/advanced/document-store/extend.md:171]:
```suggestion
To support the list-like feature, the list-like APIs should perform flag checking only when the offset2id structure is called as follows:
```

File: docs/advanced/document-store/extend.md [hunk 7]
```diff
@@ -203,6 +225,17 @@ class BackendMixin(BaseBackendMixin):
...
+```python
+@dataclass
+class MyDocStoreConfig:
+    config1: str
+    config2: str
+    list_like: bool
+    config3: Dict
+    ...
+```
+By default, it is set to be `True`.
```
Comment (alexcg1) [inline docs/advanced/document-store/extend.md:238]:
```suggestion
By default, it is set to `True`. 
```

**Review by alexcg1 — CHANGES_REQUESTED** [review]

File: docs/advanced/document-store/extend.md [hunk 8]
```diff
@@ -84,7 +84,22 @@ You will need to implement the above five functions, which correspond to the log
 
 Note that DocumentArray maintains an `offset2ids` mapping to allow a list-like behaviour. This mapping is 
 inherited from the `BaseGetSetDelMixin`. Therefore, you need to implement methods to persist this mapping, in case you 
-want to also persist the ordering of Documents inside the storage.
+want to also persist the ordering of Documents inside the storage. However, the list-like
+structure implemented by Offset2id can introduce performance bottlenecks, so you can disable this feature by passing a flag when constructing the backend.
```
Comment (JohannesMessner) [inline docs/advanced/document-store/extend.md:88]:
```suggestion
structure implemented by Offset2id can introduce performance bottlenecks, so you can disable this feature by passing a flag when constructing the backend.

In step 4 you will see how you should read in this flag (`list_like`) from the user. Here you have to use it do adapt some operations.
```

**Review by JoanFM — APPROVED** [review]

**Pull request merged**